### ISSUE TITLE
Support encoding of segmentations for non-volumetric modalities

### ISSUE THREAD

**Issue by fedorov** [QIICR/dcmqi#319]
Currently, conversion of segmentations for input datasets such as X-ray (CR) will fail due to various attributes that converters expects to be present.

For the planar images, instead of expecting that frame of reference and ImagePosition/Orientation are available, SEG should just reference the source image and not populate those attributes.

**Comment by michaelonken** [comment]
If there is something that you see in dcmseg or related DCMTK modules let me know Andrey.

**Comment by fedorov** [comment]
This issue was reported by Felix Nensa @luckfamousa.

As I thought about this, here's a complication I realized.

CR does not contain patient orientation/position. However, at the same time, image orientation/geometry attributes are mandatory for ITK volumetric formats, and ITK volumetric formats are the input format feeding segmentations into the converter, and the output format when we convert from DICOM SEG.

I can't think of a way how to ensure reliably that the geometry of the segmentation image produced by dcmqi (in ITK format) is consistent with that of the CR modality, since CR modality load/conversion will be done by another tool. Need to discuss this.

**Comment by fedorov** [comment]
Here's a "solution" to this problem I just came up with - for planimetric DICOM objects, we could always store segmentations in planimetric non-DICOM formats, such as PNG.

**Comment by pieper** [comment]
I don't think it would be a problem to have a 2D image with identity geometry at the ITK level.  Putting the segmentation back into DICOM will require some knowledge of the referenced image and from that it can handle the case of a CR or similar images.

**Comment by fedorov** [comment]
> I don't think it would be a problem to have a 2D image with identity geometry at the ITK level.

The problem is not that. The problem is that the user may load image and segmentation (which went through different reader pipelines), and the result will not produce an overlap/overlay of the segmentation over image. If you don't know anything about geometric position of the 2d slice in space, I don't see how it helps to initialize IPP/IOP.

**Comment by pieper** [comment]
From the Slicer point of view we tend to insist that images are positioned in patient space, but if you load a single slice 2D image it is just in pixel space with identity transform and unit spacing.  

Since CR and seg made from them are required to align in pixel space (same number of rows and columns) they will overlay correctly.  That's the way I understood the references that David provided.

**Comment by fedorov** [comment]
The way I see it, here is the difference of the two approaches:

Approach 1: we have no position orientation in CR, but we hard-code position/orientation in the converted DICOM SEG by saving it as NRRD or such: we make the assumption that the software loading the two will have the same conventions as Slicer

Approach 2: we have no position orientation in CR, AND we have no position/orientation in the converted DICOM SEG by saving it as PNG or such: we make the assumption that the software loading the two will treat two images with no information about position/orientation consistently

I personally think that the assumption in Approach 2 is safer.

Real problem with Approach 2 is that we have no means to save pixel spacing with PNG or such, and CR has `ImagerPixelSpacing` which is discussed in length here http://gdcm.sourceforge.net/wiki/index.php/Imager_Pixel_Spacing). So that indeed might be a sufficient reason to abandon approach 2.

```
(0018,1164) DS [0.144\0.144]                            #  12, 2 ImagerPixelSpacing
```
